libs/domain/user/address-list/address-list.component.spec.ts

```diff
@@ -28,96 +61,225 @@ describe('UserAddressListComponent', () => {
-        html`<oryx-user-address-list
-          @oryx.select=${callback}
-          .options=${{ selectable: true }}
-        ></oryx-user-address-list>`
-      );
+      service.getAddresses.mockReturnValue(of(mockAddresses));
     });
 
-    // TODO: consider dropping this unless this is a feature we need; I'd rather
-    // extended the client model and keep the state in the service if this
-    it.skip('should emit select event', () => {
-      expect(callback).toHaveBeenCalledWith(
-        expect.objectContaining({ detail: { address: expect.any(Object) } })
-      );
+    describe('and the selectable option = false', () => {
+      beforeEach(async () => {
+        element = await fixture(
+          html`<oryx-user-address-list
+            .options=${{ selectable: false }}
+          ></oryx-user-address-list>`
+        );
+      });
+
+      it('should not render an empty slot', async () => {
+        expect(element).not.toContainElement('slot[name="empty"]');
+      });
+
+      it('should render t4o address items', () => {
+        const items = element.renderRoot.querySelectorAll(
+          'oryx-tile oryx-user-address-list-item'
+        );
+        expect(items.length).toBe(2);
+      });
+
+      it('should not render an input for the items', () => {
+        expect(element).not.toContainElement('input');
+      });
     });
 
-    describe('and selected address was changed', () => {
+    describe('and the selectable option = true', () => {
       beforeEach(async () => {
-        (
-          element.renderRoot.querySelector(
-            'oryx-tile:nth-child(2) input'
-          ) as HTMLInputElement
-        ).dispatchEvent(new InputEvent('input'));
-
-        element.requestUpdate();
-        await elementUpdated(element);
+        element = await fixture(
+          html`<oryx-user-address-list
+            .options=${{ selectable: true }}
+          ></oryx-user-address-list>`
+        );
+      });
+
+      it('should render an input for the items', () => {
+        expect(element).toContainElement(`input[value='billing']`);
+        expect(element).toContainElement(`input[value='shipping']`);
+      });
+
+      it('should not select any inputs', () => {
+        expect(element).not.toContainElement('input[checked]');
+      });
+
+      describe('and when an address is selected', () => {
+        beforeEach(async () => {
+          addressStateService.get.mockReturnValue(of({ selected: 'shipping' }));
+          element = await fixture(
+            html`<oryx-user-address-list
+              .options=${{ selectable: true }}
+            ></oryx-user-address-list>`
+          );
+        });
+
+        it('should select the tile', () => {
+          expect(element).toContainElement(
+            `oryx-tile[selected] input[value='shipping']`
+          );
+        });
+      });
+
+      describe('and when an address is selected', () => {
+        beforeEach(() => {
+          element.dispatchEvent = vi.fn();
```

Comment:
```suggestion
          spy = vi.spyOn(element, 'dispatchEvent');
```

Better to spy on the method then override it